File: Guigna-Swift/Guigna/GuignaRepos.swift
import Foundation

class GRepo: GScrape {
}


class Native: GRepo {
    
    init(agent: GAgent) {
        super.init(name: "Native Installers", agent: agent)
        homepage = "http://github.com/gui-dos/Guigna/"
        itemsPerPage = 250
        cmd = "installer"
    }
    
    override func refresh() {
        var pkgs = [GItem]()
        let url = NSURL(string: "https://docs.google.com/spreadsheet/ccc?key=0AryutUy3rKnHdHp3MFdabGh6aFVnYnpnUi1mY2E2N0E")
        let xmlDoc = NSXMLDocument(contentsOfURL: url, options: Int(NSXMLDocumentTidyHTML), error: nil)
        var nodes = xmlDoc.rootElement().nodesForXPath("//table[@id=\"tblMain\"]//tr", error: nil) as [NSXMLNode]
        for node in nodes {
            if node.attribute("class") != nil { // class is not empty ('rShim')
                continue
            }
            let columns = node["td"]
            let name = columns[1].stringValue!
            let version = columns[2].stringValue!
            let homepage = columns[4].stringValue!
            let url = columns[5].stringValue!
            var pkg = GItem(name: name, version: version, source: self, status: .Available)
            pkg.homepage = homepage
            pkg.description = url
            pkg.URL = url
            pkgs += pkg
        }
        items = pkgs
    }
}

File: Guigna-RubyMotion/app/GuignaRepos.rb
# require 'GAdditions'
# require 'GuignaScrapes'
# require 'GAgent'

class GRepo < GScrape
  def update
  end
end


class Native < GRepo
  
  def initialize(agent=nil)
    super("Native Installers", agent)
    @homepage = "http://github.com/gui-dos/Guigna/"
    @items_per_page = 250
    @cmd = "installer"
  end
  
  def refresh
    pkgs = []
    url = "https://docs.google.com/spreadsheet/ccc?key=0AryutUy3rKnHdHp3MFdabGh6aFVnYnpnUi1mY2E2N0E"
    nodes = agent.nodes_for_url(url, xpath:"//table[@id=\"tblMain\"]//tr")
    nodes.each do |node|
      next if !node.attributeForName("class").nil? # class is not empty ('rShim')
      columns = node["./td"]
      name = columns[1].stringValue
      version = columns[2].stringValue
      homepage = columns[4].stringValue
      url = columns[5].stringValue
      pkg = GItem.new(name, version, self, :available)
      pkg.homepage = homepage
      pkg.description = url
      pkg.url = url
      pkgs << pkg
    end
    @items = pkgs
  end
  
end



